### PR TITLE
INI: fixed sections and ordering

### PR DESCRIPTION
**Pull request by tom-wa**
Works as default storage now + all unit tests are working.
the only issue left is the metadata added by the plugin used for ordering.

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build ini please

**Comment by markus2330** [comment]
jenkins build ini please

**Comment by markus2330** [comment]
Thank you, that is great!

Can you resolve the conflicts for the merge?

File: src/plugins/ini/ini.c [hunk 1]
```diff
@@ -395,6 +334,67 @@ static int iniKeyToElektraArray (CallbackHandle * handle, Key * existingKey, Key
 	return 1;
 }
 
+static void insertKeyIntoKeySet (Key * parentKey, Key * key, KeySet * ks)
```
Comment (markus2330) [inline src/plugins/ini/ini.c]: please add some comments for every function.

File: src/plugins/ini/ini.c [hunk 2]
```diff
@@ -580,7 +593,7 @@ int elektraIniOpen (Plugin * handle, Key * parentKey ELEKTRA_UNUSED)
 	Key * contStringKey = ksLookupByName (config, "/linecont", KDB_O_NONE);
 	if (!contStringKey)
 	{
-		pluginConfig->continuationString = strdup ("\\");
+		pluginConfig->continuationString = strdup ("\t");
```
Comment (markus2330) [inline src/plugins/ini/ini.c]: Please document this in README.md. There it says any whitespace is ok?

File: src/plugins/ini/ini.c [hunk 3]
```diff
@@ -838,7 +856,7 @@ int elektraIniGet (Plugin * handle, KeySet * returned, Key * parentKey)
 }
 
 // TODO: # and ; comments get mixed up, patch inih to differentiate and
-// create comment keys instead of writing meta data. Writing the meta
+// create comment keys instead of writing meta data. iriting the meta
```
Comment (markus2330) [inline src/plugins/ini/ini.c]: Seems to be a wrong edit.

**Comment by markus2330** [comment]
@tom-wa: please fix merge conflicts, thank you!

**Comment by tom-wa** [comment]
@markus2330 done, not sure about the one failed testcase though, does the buildserver run the checks with ini or dump as default storage ?
using ini as default storage a few others fail, because there's still the ini/key, parent and order metadata attached to the keys for ordering/sorting/inserting purpose

**Comment by markus2330** [comment]
The build run that happens automatically is done with dump as default. So  testscr_check_export should be fixed.

**Comment by markus2330** [comment]
jenkins build ini please

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
Does not round-trip properly with the default settings:

```-------- ini -----------
2c2,3
< key/subkey = another value
---
> [key]
> subkey = another value
error: Export file again_two_value.ini was not equal
```

Could you please either change `again_two_value.ini` or (my preference) the default behaviour of INI do not change the content of files by default?

**Comment by markus2330** [comment]
Thank you, these are great fixes!

**Pull request merged**